crates/epaint/src/text/text_layout.rs

```diff
@@ -943,6 +943,27 @@ impl RowBreakCandidates {
                 .or(self.any)
         }
     }
+
+    fn forget_before_idx(&mut self, index: usize) {
+        if self.space.map(|s| s < index).unwrap_or_default() {
+            self.space = None;
+        }
+        if self.cjk.map(|s| s < index).unwrap_or_default() {
+            self.cjk = None;
+        }
+        if self.pre_cjk.map(|s| s < index).unwrap_or_default() {
+            self.pre_cjk = None;
+        }
+        if self.dash.map(|s| s < index).unwrap_or_default() {
+            self.dash = None;
+        }
+        if self.punctuation.map(|s| s < index).unwrap_or_default() {
+            self.punctuation = None;
+        }
+        if self.any.map(|s| s < index).unwrap_or_default() {
+            self.any = None;
+        }
+    }
```

Comment:
This is a good trick to ensure we don't miss any member (if we add more in the future)

```suggestion
    fn forget_before_idx(&mut self, index: usize) {
        let Self {
            space,
            cjk,
            pre_cjk,
            dash,
            punctuation,
            any,
        } = self;
        if space.map_or(false, |s| s < index) {
            *space = None;
        }
        if cjk.map_or(false, |s| s < index) {
            *cjk = None;
        }
        if pre_cjk.map_or(false, |s| s < index) {
            *pre_cjk = None;
        }
        if dash.map_or(false, |s| s < index) {
            *dash = None;
        }
        if punctuation.map_or(false, |s| s < index) {
            *punctuation = None;
        }
        if any.map_or(false, |s| s < index) {
            *any = None;
        }
    }
```